### ISSUE TITLE
Using AddToBatchAsync for getting multiple list items in a single batch causes unnecessary extra rest calls

### ISSUE THREAD

**Issue by luciusbv** [pnp/pnpcore#761]
## Category

- [x] Bug

## Describe the bug

Trying to get multiple list items from a List instance having the Fields collection already loaded in memory should require a single batch request to be sent to the server. Currently for each sub-query in batch a request for list fields is sent additionally

## Steps to reproduce
```
//A list instance is created and the fields collection for it is loaded as well
IList changedList = await pnpContext.Web.Lists.GetByIdAsync(listId, p => p.Id, p => p.Title, a => a.Fields.QueryProperties(f => f.InternalName, f => f.TypeAsString, f => f.FieldTypeKind, f => f.Title));
...
//List<IChangeItem> changes contains a set of changes detected on the list, we are trying to load the list items for each changes next
for (var index = 0; index < changes.Count; index++)
{
		var changeItem = changes[index];
		result[index] = await changedList.Items.Where(a => a.Id == changeItem.ItemId).AsBatchAsync();
}
await pnpContext.ExecuteAsync();

```
if the changes set contains 14 item ids, 15 server requests will be sent:
- the actual batch for getting the list items
- 14 identical requests to dynamically load the fields for the list will be sent
![image](https://user-images.githubusercontent.com/45073117/153675163-fb54e834-2d71-4711-87ff-9dee8a942514.png)

## Expected behavior

Since the Fields are already available in memory, they should not be dynamically loaded again. Only the first batch request from the above screenshot should be issued.

## Environment details (*development & target environment*)

- **SDK version**: [1.5.0]
- **OS**: [e.g. Windows 10]
- **SDK used in**: [Azure function]
- **Framework**: [NET Core v3.1 ]
- **Browser(s)**: [N/A]
- **Tooling**: [ Visual Studio 2019]
- **Additional details**: *Check next section details.*

### Additional context
As in https://github.com/pnp/pnpcore/issues/760 the cause seem to be a mutability issue, but a different one.
![image](https://user-images.githubusercontent.com/45073117/153676210-665d5214-1ca7-4946-8831-8f23a78ee3ae.png)
The code in the screenshot is called from the AsBatchAsync() extension method.
Contrary to what the code comment is saying, EntityManager.ReplicateParentHierarchy() method does not creates a clone of the list instance which in the screenshot is stored in the "Parent" variable but a very shallow replica of it which does not contains the Fields collection.
This is why Fields collection is asked again from the server.
Possible fix: EntityManger.ReplicateParent() method should actually replicate the whole state of the list instance. Right now only the Key and Metadata are replicated.

**Comment by luciusbv** [comment]
This is also a follow up on discussion from https://github.com/pnp/pnpcore/issues/449

**Comment by jansenbe** [comment]
@luciusbv : the fix for #760 also solves this as when you called getchanges the fields collection was cleared and in the subsequent item loads the code ensured the fields where loaded. Given these subsequent requests where batched it meant that each time the code wanted to load the fields. If you still see issues with this, then please re-open.
